internal/gengapic/genrest.go

```diff
@@ -56,62 +80,515 @@ func (g *generator) genRESTMethods(serv *descriptor.ServiceDescriptorProto, serv
+	p(`return nil, "", fmt.Errorf("%s not yet supported for REST clients")`, m.GetName())
+	p("}")
+	p("")
+	p("fetch := func(pageSize int, pageToken string) (string, error) {")
+	p("  items, nextPageToken, err := it.InternalFetch(pageSize, pageToken)")
+	p("  if err != nil {")
+	p(`    return "", err`)
+	p("  }")
+	p("  it.items = append(it.items, items...)")
+	p("  return nextPageToken, nil")
+	p("}")
+	p("")
+	p("it.pageInfo, it.nextFunc = iterator.NewPageInfo(fetch, it.bufLen, it.takeBuf)")
+	p("it.pageInfo.MaxSize = int(req.GetPageSize())")
+	p("it.pageInfo.Token = req.GetPageToken()")
+	p("")
+	p("return it")
+	p("}")
+
+	return nil
+}
+
+func (g *generator) lroRESTCall(servName string, m *descriptor.MethodDescriptorProto) error {
+	lowcaseServName := lowcaseRestClientName(servName)
+	p := g.printf
+	inType := g.descInfo.Type[m.GetInputType()].(*descriptor.DescriptorProto)
+	// outType := g.descInfo.Type[m.GetOutputType()].(*descriptor.DescriptorProto)
+
+	inSpec, err := g.descInfo.ImportSpec(inType)
+	if err != nil {
+		return err
+	}
+
+	// outSpec, err := g.descInfo.ImportSpec(outType)
+	// if err != nil {
+	// 	return err
+	// }
+
+	lroType := lroTypeName(m.GetName())
+	p("func (c *%s) %s(ctx context.Context, req *%s.%s, opts ...gax.CallOption) (*%s, error) {",
+		lowcaseServName, m.GetName(), inSpec.Name, inType.GetName(), lroType)
+	p(`    return nil, fmt.Errorf("%s not yet supported for REST clients")`, m.GetName())
+	p("}")
+	p("")
+
+	p("// %[1]s returns a new %[1]s from a given name.", lroType)
+	p("// The name must be that of a previously created %s, possibly from a different process.", lroType)
+	p("func (c *%s) %[2]s(name string) *%[2]s {", lowcaseServName, lroType)
+	p("  return &%s{", lroType)
+	// TODO(dovs): return a non-empty and useful object
+	p("  }")
+	p("}")
+	p("")
+
+	g.imports[pbinfo.ImportSpec{Name: "longrunningpb", Path: "google.golang.org/genproto/googleapis/longrunning"}] = true
+
+	return nil
+}
+
+func (g *generator) emptyUnaryRESTCall(servName string, m *descriptor.MethodDescriptorProto) error {
+	info, err := getHTTPInfo(m)
+	if err != nil {
+		return err
+	}
+	if info == nil {
+		return errors.E(nil, "method has no http info: %s", m.GetName())
+	}
+
+	inType := g.descInfo.Type[m.GetInputType()]
+	inSpec, err := g.descInfo.ImportSpec(inType)
+	if err != nil {
+		return err
+	}
+
+	p := g.printf
+	lowcaseServName := lowcaseRestClientName(servName)
+	p("func (c *%s) %s(ctx context.Context, req *%s.%s, opts ...gax.CallOption) error {",
+		lowcaseServName, m.GetName(), inSpec.Name, inType.GetName())
+
+	// TODO(dovs): handle cancellation, metadata, osv.
+	// TODO(dovs): handle http headers
+	// TODO(dovs): handle deadlines
+	// TODO(dovs): handle call options
+	p("// The default (false) for the other options are fine.")
+	p("// Field names should be lowerCamel, not snake.")
+	p("m := protojson.MarshalOptions{AllowPartial: true, EmitUnpopulated: true, UseProtoNames: false}")
```

Comment:
`UseProtoNames` should default to `false` - do we need to explicitly include it here?

Reply:
I wanted to make it clear that this was a deliberate design choice. If you think it's cleaner without I'll remove it.

Reply:
I don't mind being explicit here, but that's just my opinion.